app-tracking-protection/vpn-impl/src/main/java/com/duckduckgo/mobile/android/vpn/ui/tracker_activity/DeviceShieldActivityFeedFragment.kt

```diff
@@ -83,6 +87,7 @@ class DeviceShieldActivityFeedFragment : DuckDuckGoFragment() {
             adapter = trackerFeedAdapter
         }
 
+        // let calling suspend fun define the thread
```

Comment:
It's not clear why you'd keep `lifecycleScope.launch` here instead of choosing a dispatcher... Can you elaborate?

Reply:
I can revert this change for now if you feel strongly. 
After going though all these changes, talking to some ppl and thinking about it for a long time, my high level rational behind for me would be:
* only functions/APIs that (they themselves) perform IO are declared as `suspend` and they specify the `dispatcher`
* any other function (that doesn't do themselves IO) and just call those APIs should be imperative and use `launch` without specifying any dispatcher (ie. we do not mark a function as `suspend` just because it needs to call a suspend function)
  * _Note: I know we currently have cases where we do that (marking a fun as `suspend` just because it calls another `suspend`) when we want to block on the result. I really want to thoroughly review those cases because I am currently not sure how much of it is "need it" vs how much of it is just "code smell"._

Tbf, our (current) usage of `suspend` is a bit "crafty". So this PR is an imperfect first attempt just focused on improving lock contention. After that my proposal would be to discuss how `suspend` should be used in the code base, and enforce as much as we can via static analysis.

Reply:
I'm not sure on the best answer here, so will first instead try give my (still forming) thoughts.

Looking back at the Asana task description, there is this "Possibly lint check to enforce explicitly picking a context when using launch" which I guess is what it all comes down to. _is it ever valid / good practice to launch without specifying a dispatcher?_. 

- If you'd asked me a month ago, I'd have said "no" 
- but since then I've noticed it feels a bit strange to be specifying a dispatcher (typically now `io`) in a place where it doesn't directly do IO, but instead is just launching because it calls a `suspend` function (your 2nd bullet point).

On the other hand, if you don't specify the dispatcher you will sometimes be using the `default` dispatchers (which this PR is trying to avoid) and sometimes be using the `main` dispatcher depending on which scope you're using. In a perfect world, all `suspend` functions doing something intensive would internally take care of ensuring it uses `io` dispatcher (ideal in my opinion, anyway) so that it shouldn't matter how it was called, but that's definitely not the case in the codebase yet and we do have to be very careful right now about how the coroutine was launched. 


So back to _is it ever valid / good practice to launch without specifying a dispatcher?_

- I think **no** when launching on a scope that would then use the `default` (which we're trying to avoid)
- Technically it would be fine to launch on an activity's `lifecycleScope` or `viewModelScope` without specifying since it wouldn't use `default`.
- But wondering if it's just simpler to be explicit at all times, rather than allowing it sometimes for some scopes but not others. It's not hard to be deliberate and specify the `main()` dispatcher if that's what you're wanting.

So no, I don't feel strongly and indeed I'm still trying to form an opinion on some of this.